### PR TITLE
Upgrade dependencies to gapic-* 0.9.x packages

### PR DESCRIPTION
**Pull request by bjwatson**
@dhermes These are the new dependencies for https://github.com/GoogleCloudPlatform/google-cloud-python/issues/2258 (and that you need for the 0.19.0 release of google-cloud-python).

Please note that we changed the names of the packages from `gax-*` to `gapic-*`. It was a mistake to call them `gax-*` (that just refers to the `google-gax` library), and should have always called them `gapic-*` instead.

Fixes https://github.com/GoogleCloudPlatform/google-cloud-python/issues/2258.

### CODE REVIEW

**Comment by bjwatson** [comment]
Thanks for updating this ticket with labels @tseaver.

I just realized that I forgot to update the import statements. Just pushed another commit to do that.

**Comment by daspecster** [comment]
@bjwatson travis says...
```
No matching distribution found for gapic-google-pubsub-v1<0.10dev,>=0.9.0
```
It's on testpypi  https://testpypi.python.org/pypi/gapic-google-pubsub-v1 is at 0.9.1,
but not the production pypi.

https://pypi.python.org/pypi/gapic-google-pubsub-v1 <-- 404's

**Comment by tseaver** [comment]
@bjwatson this change is hard for us to evaluate / test without updating the machinery to install those deps from https://testpypi.python.org/pypi.  Maybe add a temporary command to the `tox.ini` which pre-installs them from there?

**Comment by daspecster** [comment]
@bjwatson, sorry I didn't read your original comment closely.

Adding...
```
commands =
    pip install -i https://testpypi.python.org/pypi <package name>
```
to the tox environments, as @tseaver mentioned, should get this running.

**Comment by bjwatson** [comment]
I realized that by setting upper bounds on the `0.8.x` version of the `gax-google-*` packages, that I could safely push the new `0.9.x` packages to PyPI. I've just done that, and am now running these tests.

**Comment by bjwatson** [comment]
This is ready to merge once the tests pass. I'm not familiar enough with them to know why they're breaking. Could someone take a look and let me know if/what I broke?

I'll be AFK for an hour, but will check back in after that. Thanks!

**Comment by dhermes** [comment]
@bjwatson For future reference, [`tox` has support][1] for multiple index servers.

[1]: http://tox.readthedocs.io/en/stable/example/basic.html#installing-dependencies-from-multiple-pypi-servers

**Comment by dhermes** [comment]
I'm looking at the breakages now

**Comment by daspecster** [comment]
@dhermes https://travis-ci.org/GoogleCloudPlatform/google-cloud-python/builds/160481390#L855-L856

**Comment by dhermes** [comment]
I just pushed a commit to @bjwatson's branch to fix the coverage issue. Really cool new GitHub feature allowing maintainers to push commits to PRs!

@jonparrott Where to file bugs for the CLA-bot?

File: setup.py [hunk 1]
```diff
@@ -15,17 +15,17 @@
...
-    'gax-google-pubsub-v1 >= 0.8.0, < 0.9dev',
-    'grpc-google-pubsub-v1 >= 0.8.0, < 0.9dev',
-    'gax-google-logging-v2 >= 0.8.0, < 0.9dev',
-    'grpc-google-logging-v2 >= 0.8.0, < 0.9dev',
+    'grpcio >= 1.0.0',
+    'google-gax >= 0.13.0, < 0.14dev',
+    'gapic-google-pubsub-v1 >= 0.9.0, < 0.10dev',
+    'grpc-google-pubsub-v1 >= 0.9.0, < 0.10dev',
+    'gapic-google-logging-v2 >= 0.9.0, < 0.10dev',
+    'grpc-google-logging-v2 >= 0.9.0, < 0.10dev',
```
Comment (dhermes) [inline setup.py]: @bjwatson This is just defensive? `0.10` doesn't exist yet?

**Comment by dhermes** [comment]
@bjwatson This LGTM though a rebase may be important here to make sure we actually still cover all imports. Rather than a rebase you could just re-start this branch from `HEAD` locally and then `git merge --squash origin/breaking-gapic-changes` onto it. This also has the benefit of removing my commit and making CLA-bot happy.

**Comment by tseaver** [comment]
@dhermes the [`indexserver` directive docs](https://testrun.org/tox/latest/config.html#confval-indexserver) state that the directive is deprecated. :(

Also, we could have worked around the issue better in `tox.ini` before #2308:  relying on pip to find the gRPC-related dependencies implicitly makes this harder.

**Comment by dhermes** [comment]
> the `indexserver` directive docs state that the directive is deprecated. :(

Ha, my bad.

----

> Also, we could have worked around the issue better in `tox.ini` before #2308: relying on pip to find the gRPC-related dependencies implicitly makes this harder.

I'm not sure what you mean.

**Comment by tseaver** [comment]
@dhermes #2308 removed the explicit `[grpc]deps` bit in `tox.ini`:  that would have given us a place to ensure that the new dependencies were installed (in fact, without that section, the `indexserver` option would have been worthless).

**Comment by dhermes** [comment]
@tseaver How is that different than putting them in `setup.py`?

**Comment by tseaver** [comment]
In `tox.ini`, they get installed before `google-cloud-python`:  the tox dependency syntax allows us to specify more options (e.g, URLs, local files) than can be spelled in `install_requires`.  For instance, @bjwatson might have (temporarily) vendored in the sdists, and spelled installing them:`

```ini
[grpc]
deps =
    _vendor/grpc-spanner-v1-0.8.1.tar.gz
    _vendor/gapic-google-pubsub-v1-0.9.0b1.tar.gz
    _vendor/grpc-google-pubsub-v1-0.9.0b1.tar.gz
    _vendor/gapic-google-logging-v2-0.9.0b1.tar.gz
    _vendor/grpc-google-logging-v2-0.9.0b1.tar.gz
```

**Comment by bjwatson** [comment]
@dhermes I thought that tox might have that feature, but I couldn't quickly find correct incantation.

Thanks for looking into the breakages. I'm at a hack-a-thon today for gcloud-java samples, but I can be pre-empted if there's anything I need to fix in my packages. I'd like to no longer be blocking your `0.19.0` release by EOD. PLMK.

**Comment by dhermes** [comment]
I just did a squash merge. No worries.

**Comment by bjwatson** [comment]
Woo hoo!

**Comment by jonparrott** [comment]
Ignore clabot if you can manually verify all committers are under CLA.

**Comment by jonparrott** [comment]
(clabot can't handle multiple authors, it's a known issue)

File: setup.py (hunk 1 above)
Comment (bjwatson) [inline setup.py]: Yes. Not having this guard makes me nervous about pushing breaking changes to PyPI.

**Comment by bjwatson** [comment]
@dhermes Sorry, just saw your comments about squashing my commits. I didn't know that y'all kept PRs squashed. On GAPIC, we always wait until the merge-time to squash-and-commit using the commit button dropdown menu.

**Comment by dhermes** [comment]
> using the commit button dropdown menu.

1. We prefer keeping the commits from PRs in history
1. The squash commit isn't as easy to identify in history. Usually when making release notes we just go through all the merge commits created by PRs (not squashed regular commits).

**Comment by dhermes** [comment]
No worries though

**Comment by bjwatson** [comment]
I see, so you don't normally squash. This was just a one-off request due to multiple authors?

**Comment by dhermes** [comment]
Looks like this broke some system tests. Working on a fix now.

**Pull request merged**